cai_causal_graph/causal_graph.py

```diff
@@ -104,7 +104,9 @@ def __iter__(self) -> Iterator:
     def nodes(self) -> List[Node]:
         """Return a list of all nodes."""
         # Instantiate new nodes to remove any edge information.
-        return [Node(n.identifier, meta=n.meta) for n in self._graph.nodes]
+        return [
+            self._graph._NodeCls(n.identifier, meta=n.meta, variable_type=n.variable_type) for n in self._graph.nodes
```

Comment:
This is a bug fix as before `variable_type` was not being passed in so skeleton would not know it was a binary node for example. But I also changed to use the graph node class so the skeleton will have a time series node, as opposed to standard node, if it is a time series graph.